core/src/banking_stage.rs

```diff
@@ -2087,7 +2311,7 @@ impl BankingStage {
             );
 
             Self::push_unprocessed(
-                buffered_packet_batches,
+                unprocessed_transaction_storage,
```

Comment:
Would be good here to augment the `LeaderSlotMetricsTracker` to track the number of dropped votes due to being too old. This will tell us how effective this change actually is

This can be done by augmenting the leader metrics here https://github.com/taozhu-chicago/solana/blob/27f4a48a97e35431e22c13292c7ea8e7bdc67255/core/src/leader_slot_banking_stage_metrics.rs#L127.

Once this extra field is released, we can then lookup the metric `banking_stage-leader_slot_packet_counts`: https://github.com/taozhu-chicago/solana/blob/27f4a48a97e35431e22c13292c7ea8e7bdc67255/core/src/leader_slot_banking_stage_metrics.rs#L137`, filtered by `id = 1` for the tpu vote thread to see the average dropped old votes per block

Reply:
Might also be good to see how many initial votes were drained/correlate with how many packets were processed which is already available in `leader_slot_packet_counts. committed_transactions_count`: https://github.com/taozhu-chicago/solana/blob/27f4a48a97e35431e22c13292c7ea8e7bdc67255/core/src/leader_slot_banking_stage_metrics.rs#L170-L174

Reply:
On second thought, instead of augmenting the existing `banking_stage-leader_slot_packet_counts`, we just create an separate `banking_stage-vote_packet_counts` only reported by `id = 1` for these additional vote metrics. These can be reported in addition to the `banking_stage-leader_slot_packet_counts` on the thread with `id = 1`

Steps to do this:
1. Add an additional vote metrics struct here: https://github.com/taozhu-chicago/solana/blob/27f4a48a97e35431e22c13292c7ea8e7bdc67255/core/src/leader_slot_banking_stage_metrics.rs#L254 with a `report()` method like the other structs.
2. Augment this function here: https://github.com/taozhu-chicago/solana/blob/27f4a48a97e35431e22c13292c7ea8e7bdc67255/core/src/leader_slot_banking_stage_metrics.rs#L271-L276 to additionally report the vote metrics

Reply:
For those voting metrics, might also be helpful to give us additional insight that if we saw `N` votes for instance, how many of those:

1. were unique vote accounts
2. percentage of the stake that maps to those accounts

Can also be done in a separate PR 😃